python/source.py

```diff
@@ -41,13 +41,13 @@ def __init__(self, frequency, start_time=0, end_time=float('inf'), width=0, cuto
 
 class GaussianSource(SourceTime):
 
-    def __init__(self, frequency, width, start_time=0, cutoff=5.0):
+    def __init__(self, frequency, width, fwidth=True, start_time=0, cutoff=5.0):
         super(GaussianSource, self).__init__()
         self.frequency = frequency
-        self.width = width
-        self.start_time = 0
+        self.width = 1 / width if fwidth else width
```

Comment:
I would make `width=0, fwidth=0` a numeric default, and then set `self.width = 1 / fwidth if fwidth > 0 else width`, so that you can pass either `width` or `fwidth`.

(We could modify the gaussian source in the C++ code to support `width=0` by converting it to a width of the timestep Δt.)